scala/private/rule_impls.bzl

```diff
@@ -986,10 +986,12 @@ def _pack_source_jars(ctx):
         java_toolchain = find_java_toolchain(ctx, ctx.attr._java_toolchain),
         host_javabase = find_java_runtime_toolchain(ctx, ctx.attr._host_javabase),
     )
-    if scala_source_jar:
-        source_jars.append(scala_source_jar)
 
-    return source_jars
+    return scala_source_jar
+
+def _pack_source_jars(ctx):
+    source_jar = _pack_source_jar(ctx)
+    return [source_jar] if source_jar else []
```

Comment:
isn't `source_jar` always defined now? I'm confused how this can ever return `[]`. Can we add a comment if I am missing something?

Reply:
PS: if I am right, can we remove `_pack_source_jars` and and just use `_pack_source_jar`?

Reply:
the `java_common.pack_sources` may return None. @iirina maybe you know in which cases?

Reply:
@johnynek just add a comment saying that `java_common.pack_sources` may return `None`? Honestly I want to do some refactoring here as I'd like us to drop the `scala` attribute or at least significantly trim down what we're passing there to rely on JavaInfo.

Reply:
yeah, adding that comment would be fine to me, but I wonder if we really need two methods, maybe just put this logic at the end of the first call?

I would also love to remove the `scala` attribute and just use `JavaInfo`. If it doesn't break intellij, I think we should be able to do that. We should have an issue (maybe already do) to track that.

Reply:
Just now saw your comment about removing `_pack_source_jars` completely. I preferred to focus on this change since (like I wrote above) I want to consider greatly limiting the scala attribute and so probably also will be able to remove `_pack_source_jars`.

Reply:
no problem.

Reply:
`java_common.pack_sources` returns `None` if there's nothing to pack, meaning if both `source_jars` and `sources` are empty.